data-prepper-core/src/main/java/org/opensearch/dataprepper/parser/PipelineTransformer.java

```diff
@@ -125,6 +153,20 @@ private void buildPipelineFromConfiguration(
                     .map(this::newProcessor)
                     .collect(Collectors.toList());
 
+            LOG.info("Building sinks for the pipeline [{}]", pipelineName);
+            final List<DataFlowComponent<Sink>> sinks = pipelineConfiguration.getSinkPluginSettings().stream()
+                    .map(this::buildRoutedSinkOrConnector)
+                    .collect(Collectors.toList());
+
+            final List<PluginError> subPipelinePluginErrors = pluginErrorCollector.getPluginErrors()
+                    .stream().filter(pluginError -> pipelineName.equals(pluginError.getPipelineName()))
+                    .collect(Collectors.toList());
+            if (!subPipelinePluginErrors.isEmpty()) {
+                throw new InvalidPluginConfigurationException(
```

Comment:
Where is this caught and handled?

It may make sense to have a single class to take all the errors and consolidate them. Also, multiple pipelines may fail.

And is this the way we want to log it? This will likely create a single long, multi-line log. But, would it be clearer to have multiple log lines?

I'd probably throw an exception that includes a list of `PluginError` and then have a single class for handling errors which can then decide how to format and log these.

Reply:
It is actually caught and handled in L197

>I'd probably throw an exception that includes a list of PluginError and then have a single class for handling errors which can then decide how to format and log these.

Agree this part can be further refactored

Reply:
opened a separate issue to track: https://github.com/opensearch-project/data-prepper/issues/4854